src/libraries/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Windows.cs

```diff
@@ -103,6 +102,10 @@ public override ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken
                 AsyncOverSyncRead(buffer, cancellationToken);
         }
 
+        /// <summary>Initiates an async-over-sync read for a pipe opened for non-overlapped I/O.</summary>
+        private ValueTask<int> AsyncOverSyncRead(Memory<byte> buffer, CancellationToken cancellationToken) =>
+            AsyncOverSyncWithIoCancellation.InvokeAsync(static s => s.Stream.ReadCore(s.Buffer.Span), (Stream: this, Buffer: buffer), cancellationToken);
```

Comment:
You don't appear to use `Buffer` in your value tuple. Wanna just pass `this` instead?